pkg/diagnose/scan.go

```diff
@@ -234,7 +233,7 @@ func Scan(ctx context.Context, opts ...OpOption) error {
 			query_log_tail.WithDedup(true),
 			query_log_tail.WithCommands(pkg_dmesg.DefaultDmesgScanCommands),
 			query_log_tail.WithLinesToTail(op.lines),
-			query_log_tail.WithSelectFilter(dmesg.DefaultDmesgFiltersForNvidia()...),
+			query_log_tail.WithSelectFilter(defaultDmesgFilters()...),
```

Comment:
We should refactor the dmesgCheck feature. It's repetitive across other components, even though it composes these components here.

IMO we should call the existing dmesg check components directly or their check functions to determine if the log line is matched, we should hide some logic here ,for example we should use https://github.com/leptonai/gpud/blob/2175d7d3671ccde724a6145ddbb87aa3f36e2cbd/pkg/nvidia-query/xid/dmesg/dmesg.go#L61 instead of line 247..